### PR TITLE
feat: [CDS-34542]: Add tooltips for artifactory repo name and file path

### PR DESCRIPTION
**Pull request by willd0yle**
##### Summary:
Add tooltip to explain what is required for the artifact and repo name form elements. Artifact path uses the remote tool tip that will link to documentation that explains in more detail on what is needed for the input.

##### Jira Links:

[](https://harness.atlassian.net/browse/CDS-34542)

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by willd0yle** [comment]
retrigger jest

**Review by chrisblakely01 — APPROVED** [review]

**Review by kylekstewart — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx [hunk 1]
```diff
@@ -193,25 +194,43 @@ export const TFArtifactoryForm: React.FC<StepProps<any> & TFArtifactoryProps> =
...
+                    }}
+                  />
+                </Text>
                 <div className={cx(stepCss.formGroup, stepCss.md)}>
                   {getMultiTypeFromValue(connectorRef) === MultiTypeInputType.FIXED ? (
                     <FormInput.MultiTypeInput
                       selectItems={connectorRepos ? connectorRepos : []}
                       name={tfArtifactoryFormInputNames(isConfig).repositoryName}
-                      label={getString('pipelineSteps.repoName')}
+                      label={''}
```
Comment (YounixIO) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx:221]: Is label a mandatory prop? If not, can we remove the label prop?

Comment (willd0yle) [reply src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx:221]: It is a mandatory prop unfortunately

**Review by YounixIO — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx [hunk 2]
```diff
@@ -238,9 +257,14 @@ export const TFArtifactoryForm: React.FC<StepProps<any> & TFArtifactoryProps> =
...
                   <MultiTypeFieldSelector
                     name={tfArtifactoryFormInputNames(isConfig).artifactPaths}
-                    label={getString(isConfig ? 'pipeline.artifactPathLabel' : 'cd.artifactPaths')}
                     style={{ width: 370 }}
                     allowedTypes={allowableTypes.filter(item => item !== MultiTypeInputType.EXPRESSION)}
+                    label={
+                      <Text flex={{ inline: true }}>
+                        {getString(isConfig ? 'pipeline.artifactPathLabel' : 'cd.artifactPaths')}
+                        <HarnessDocTooltip useStandAlone={true} tooltipId="artifactory_file_path" />
+                      </Text>
```
Comment (knagurski) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx:266]: You can actually simplify this and drop the `HarnessDocTooltip`
```suggestion
                      <Text
                        flex={{ inline: true }}
                        tooltipProps={{ tooltipId: 'artifactory_file_path' }}
                      >
                        {getString(isConfig ? 'pipeline.artifactPathLabel' : 'cd.artifactPaths')}
                      </Text>
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformVarfile.module.scss [hunk 3]
```diff
@@ -296,3 +296,7 @@
 .newConnectorButton {
   margin-top: 40px !important;
 }
+
+.tooltipProps {
```
Comment (knagurski) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformVarfile.module.scss:300]: Not a great name. Maybe something more descriptive
```suggestion
.repoNameButtonTooltip {
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx [hunk 4]
```diff
@@ -193,25 +194,43 @@ export const TFArtifactoryForm: React.FC<StepProps<any> & TFArtifactoryProps> =
...
+                      usePortal: false,
+                      isDark: true,
+                      className: css.tooltipProps
+                    }}
+                    iconProps={{
+                      size: 12,
+                      padding: { right: 'small', top: 'small', bottom: 'small' }
+                    }}
+                  />
+                </Text>
```
Comment (knagurski) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx:215]: Would it make more sense to put a tooltip on the `Text` component and use the `iconRight` prop to specify that you want the tooltip icon?

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx (hunk 1 above)
Comment (knagurski) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx:221]: Might be blank, but it's still a string, so you can drop the parens (same below)
```suggestion
                      label=""
```

**Review by knagurski — COMMENTED** [review]
Nothing wrong but there's a few things that could be improved and simplified

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**